libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/helpers/logfile/LogFileWriter.kt

```diff
@@ -0,0 +1,45 @@
+package org.wordpress.android.util.helpers.logfile
+
+import org.jetbrains.annotations.TestOnly
+import java.io.File
+import java.io.FileWriter
+import java.util.Date
+import org.wordpress.android.util.DateTimeUtils
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+/**
+ * A class that manages writing to a log file.
+ *
+ * This class creates and writes to a log file, and will typically persist for the entire lifecycle
+ * of its host application.
+ */
+class LogFileWriter @JvmOverloads constructor(
+    logFileProvider: LogFileProviderInterface,
+    fileId: String = DateTimeUtils.iso8601FromDate(Date())
+) {
+    private val file = File(logFileProvider.getLogFileDirectory(), "$fileId.log")
+    private val fileWriter: FileWriter = FileWriter(file)
+
+    /**
+     * An serial executor used to write to the file in a background thread
+     */
+    val queue: ExecutorService = Executors.newSingleThreadExecutor()
```

Comment:
Hmm.. I am unfamiliar with Executors and didn't really think about this as a solution to our async problem. I looked up the documentation and looks like this will work as we expect to, so 👍 

Couple issues though:
1. We probably don't want to expose this queue to outside, so should we make it `private`?
2. While we are at that, how do you feel about moving this to the constructor so the caller gets to decide what kind of executor they'll use? We can still keep the `Executors.newSingleThreadExecutor()` as the default value, but this will give us a way to alter it in the app or in the test. If we do this, we might want to have a different name than `queue`.
3. The documentation probably should say `A serial executor` instead of `An serial executor` 🙈

Reply:
I'm onboard with changing the queue to `private` for sure, that makes sense.

Fixed the typo too (you can tell I added the `serial` part after)

For the constructor thing, I think this is an important part of the implementation – a non-serial queue would have race conditions that could end up with parts of the file written in the wrong order, which for this purpose would be very bad.

I considered the idea of an `ImmediateExecutor` that could be mocked in to replace the default, but that would again change the behaviour of the class – skipping the async bits in a test case would be nice, but wouldn't be representative of real-world circumstances – we expect there to be a delay in writes. To keep the tests aligned with how the class works in real use, I think it makes sense to leave this as-is unless we come up with a good use-case for changing it?

I updated the other things in 072f22a9982313a1e8b04034f31a09b322999592